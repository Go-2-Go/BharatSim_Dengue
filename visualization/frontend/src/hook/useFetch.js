import {useEffect, useState} from "react";
import fetch from "../utils/fetch";
import {httpMethods} from "../constants/httpMethods";

const useFetch = (url, method=httpMethods.GET) => {
  const [data, setData] = useState();

  useEffect( () => {
    const fetchData = async () => {
      const resData = await fetch({url,method});
      setData(resData)
    }
    fetchData();
  },[])

  return data;
}

export default useFetch;