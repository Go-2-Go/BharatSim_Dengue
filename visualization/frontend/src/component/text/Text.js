import React from "react";

import {url} from "../../utils/url";

import useFetch from "../../hook/useFetch";

const Text = () => {
  const text = useFetch(url.HELLO);
  return <div>{text}</div>
}

export default Text;