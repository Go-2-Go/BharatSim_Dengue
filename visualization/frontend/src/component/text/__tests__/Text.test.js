import { render } from '@testing-library/react';
import React from 'react';

import Text from '../Text';
import useFetch from '../../../hook/useFetch';

jest.mock('../../../hook/useFetch');

describe('<Text />', () => {
  beforeEach(() => {
    useFetch.mockReturnValue('hello, Welcome to bharatsim');
  });

  it('should have fetched text in <Text /> component', () => {
    const { getByText } = render(<Text />);

    const element = getByText(/hello, Welcome to bharatsim/i);
    expect(element).toBeInTheDocument();
  });

  it('should match a snapshot for <Text />', () => {
    const { container } = render(<Text />);

    expect(container).toMatchSnapshot();
  });
});
