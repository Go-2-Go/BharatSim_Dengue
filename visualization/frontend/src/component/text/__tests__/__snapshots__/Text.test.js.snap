// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match a snapshot for <Text /> 1`] = `
<div>
  <div>
    hello, Welcome to bharatsim
  </div>
</div>
`;
